unity-renderer/Assets/Scripts/MainScripts/DCL/Controllers/HUD/ProfileHUD/ProfileHUDViewV2.cs

```diff
@@ -0,0 +1,424 @@
+
+    [SerializeField]
+    internal GameObject nonConnectedWalletSection;
+
+    [Header("Thumbnail")]
+    [SerializeField]
+    internal RawImage imageAvatarThumbnail;
+
+    [SerializeField]
+    protected internal Button buttonToggleMenu;
+
+    [Header("Texts")]
+    [SerializeField] internal TextMeshProUGUI textName;
+    [SerializeField] internal TextMeshProUGUI textPostfix;
+    [SerializeField] internal TextMeshProUGUI textAddress;
+
+    [Header("Buttons")]
+    [SerializeField] protected internal Button buttonClaimName;
+    [SerializeField] protected internal Button buttonCopyAddress;
+    [SerializeField] protected internal Button buttonLogOut;
+    [SerializeField] protected internal Button buttonSignUp;
+    [SerializeField] protected internal Button_OnPointerDown buttonTermsOfServiceForConnectedWallets;
+    [SerializeField] protected internal Button_OnPointerDown buttonPrivacyPolicyForConnectedWallets;
+    [SerializeField] protected internal Button_OnPointerDown buttonTermsOfServiceForNonConnectedWallets;
+    [SerializeField] protected internal Button_OnPointerDown buttonPrivacyPolicyForNonConnectedWallets;
+
+    [Header("Name Edition")]
+    [SerializeField] protected internal Button_OnPointerDown buttonEditName;
+    [SerializeField] protected internal Button_OnPointerDown buttonEditNamePrefix;
+    [SerializeField] internal TMP_InputField inputName;
+    [SerializeField] internal TextMeshProUGUI textCharLimit;
+    [SerializeField] internal ManaCounterView manaCounterView;
+    [SerializeField] internal ManaCounterView polygonManaCounterView;
+
+    [Header("Tutorial Config")]
+    [SerializeField] internal RectTransform tutorialTooltipReference;
+
+    [Header("Description")]
+    [SerializeField] internal GameObject charLimitDescriptionContainer;
+    [SerializeField] internal GameObject descriptionStartEditingGo;
+    [SerializeField] internal GameObject descriptionIsEditingGo;
+    [SerializeField] internal Button descriptionStartEditingButton;
+    [SerializeField] internal Button descriptionIsEditingButton;
+    [SerializeField] internal TMP_InputField descriptionInputText;
+    [SerializeField] internal TextMeshProUGUI textCharLimitDescription;
+
+    [SerializeField] internal GameObject descriptionContainer;
+
+    private InputAction_Trigger.Triggered closeActionDelegate;
+
+    private Coroutine copyToastRoutine = null;
+    private UserProfile profile = null;
+    private Regex nameRegex = null;
+    private string description;
+    private string userId;
+    private StringVariable currentPlayerId;
+
+    public event EventHandler ClaimNamePressed;
+    public event EventHandler SignedUpPressed;
+    public event EventHandler LogedOutPressed;
+    public event EventHandler WalletTermsAndServicesPressed;
+    public event EventHandler WalletPrivacyPolicyPressed;
+    public event EventHandler NonWalletTermsAndServicesPressed;
+    public event EventHandler NonWalletPrivacyPolicyPressed;
```

Comment:
This event handler are never used, why? they should behave the same as the old panel

Reply:
The old system does not differentiate between nonWallet and wallet, I modified the events to match the interface